search/includes/classes/class-health.php

```diff
@@ -595,14 +595,27 @@ public static function filter_expected_post_rows( $rows, $post_types, $post_stat
 	}
 
 	public static function simplified_diff_document_and_prepared_document( $document, $prepared_document ) {
+		$checked_keys = [];
+
 		foreach ( $document as $key => $value ) {
+			$checked_keys[ $key ] = true;
 			if ( in_array( $key, self::DOCUMENT_IGNORED_KEYS, true ) ) {
 				continue;
 			}
 
 			if ( is_array( $value ) ) {
-				$recursive_diff = self::simplified_diff_document_and_prepared_document( $document[ $key ], $prepared_document[ $key ] );
```

Comment:
Cha turns out the simple one used to be very shallow too :) The `$recursive_diff` was ignored before.